apps/coscrad-frontend/src/components/notes/note-index.presenter.tsx

```diff
@@ -1,18 +1,92 @@
-import { INoteViewModel } from '@coscrad/api-interfaces';
-import { Link } from 'react-router-dom';
+import {
+    EdgeConnectionMemberRole,
+    EdgeConnectionType,
+    IEdgeConnectionMember,
+    INoteViewModel,
+} from '@coscrad/api-interfaces';
+import { HeadingLabel, IndexTable } from '../../utils/generic-components/presenters/tables';
+import { CellRenderersDefinition } from '../../utils/generic-components/presenters/tables/generic-index-table-presenter/types/cell-renderers-definition';
 import { HasData } from '../higher-order-components';
+import { renderAggregateIdCell } from '../resources/utils/render-aggregate-id-cell';
 
-export const NoteIndexPresenter = ({ data: notes }: HasData<INoteViewModel[]>): JSX.Element => (
-    <div>
-        {notes.map((note) => (
-            <div key={note.id} data-testid={note.id}>
-                <h1>
-                    note: <Link to={note.id}>{note.id}</Link>
-                </h1>
-                <p>{note.note}</p>
-                <h2>Full Props (stringified):</h2>
-                <p>{JSON.stringify(note)}</p>
+interface DisplayConnectedResourcesInfoProps {
+    resourceInfos: IEdgeConnectionMember[];
+}
+
+const DisplayConnectedResourcesInfo = ({
+    resourceInfos,
+}: DisplayConnectedResourcesInfoProps): JSX.Element => {
+    if (resourceInfos.length === 1) {
+        const {
+            compositeIdentifier: { type, id },
+        } = resourceInfos[0];
+
+        return (
+            <div>
+                A note about {type}/{id}
             </div>
-        ))}
-    </div>
-);
+        );
+    }
+
+    const {
```

Comment:
These two `const`s are tricky somehow.  Is there a way to name them more explicitly?  Just reaching here...

Reply:
Here's what I came up with:
```ts
const DisplayConnectedResourcesInfo = ({
    resourceInfos,
    connectionType,
}: DisplayConnectedResourcesInfoProps): JSX.Element => {
    if (connectionType === EdgeConnectionType.self) {
        const { compositeIdentifier } = resourceInfos[0];

        return <div>A note about {formatCompositeIentifier(compositeIdentifier)}</div>;
    }

    const [fromMember, toMember] = sortEdgeConnectionMembers(resourceInfos);

    const fromMessage = `connection from ${formatCompositeIentifier(
        fromMember.compositeIdentifier
    )} to ${toMember.compositeIdentifier}`;

    return <div>{fromMessage}</div>;
};
```